### ISSUE TITLE
Mending Touch will absolutely obliterate you with Pacifism 

### ISSUE THREAD

**Issue by TetrakoHateThis** [tgstation/tgstation#84557]
## Issue Summary

Oh, yeah, this is an issue we noticed from DOWNSTREAM. We figured it out a bit more, and it's not anything downstream wise that's causing the issue.

Utilizing Mending Touch with the Pacifism Quirk can cause a much higher multiplier than is expected, allowing someone to basically kill themselves on accident, for no tangible gain. Tested this in an active round, used mending touch on someone with 40 burn damage, which promptly killed me by dealing 370+ burn damage, husking me too. This is roughly a 9x modifier. 

EDIT: We did some checking and it definitely ramps up damage and kills you with solely Pacifism.

## Round ID:
[11153](https://scrubby.melonmesa.com/round/11153)

## Testmerges:

I accidentally did this report on TG's git before Nova, but we've clarified the issue downstream and it's gotta be an interaction with the new Mending Touch and Pacifism.

## Reproduction:

Give yourself the Mending Touch mutation + Pacifism Quirk.
Find mob with roughly 40+ damage.
Use Mending Touch on mob.
Instantly Die.
![logofmendingtouchdamage](https://github.com/tgstation/tgstation/assets/81869793/a7117874-2c49-43e4-9bd5-6e2b3031df39)

**Comment by ZephyrTFA** [comment]
lol

**Comment by TetrakoHateThis** [comment]
Made some edits cuz it's definitely not a downstream issue now.

**Comment by 00-Steven** [comment]
think i got an idea as to why maybe

**Comment by Iamgoofball** [comment]
>nova

wrong repo please go to https://github.com/NovaSector/NovaSector

**Comment by TetrakoHateThis** [comment]
> > nova
> 
> wrong repo please go to https://github.com/NovaSector/NovaSector

Yes, I'm aware Goof, but we did testing and it's likely to be an upstream (here) issue since none of the downstream stuff was found to be the problem.

**Comment by 00-Steven** [comment]
goof I just read the code and then TESTED THE CODE
this is definitely about our code.

specifically each time you use it, it reapplies the multiplier
![image](https://github.com/tgstation/tgstation/assets/42909981/b73852cf-4640-42ec-b8e8-0b551883711e)
https://github.com/tgstation/tgstation/blob/e8476ae9336dcd7cccf65d89ecfaa86fa9330e4e/code/datums/mutations/touch.dm#L130-L133
https://github.com/tgstation/tgstation/blob/e8476ae9336dcd7cccf65d89ecfaa86fa9330e4e/code/datums/mutations/touch.dm#L156-L159

**Comment by 00-Steven** [comment]
There's a few other odd bits to point out in this code.
It states `heal_multiplier` "Multiplies the amount healed, without increasing the received damage.":
https://github.com/tgstation/tgstation/blob/e8476ae9336dcd7cccf65d89ecfaa86fa9330e4e/code/datums/mutations/touch.dm#L130-L133
But while it _is_ used as a direct mult on healing, which doesn't affect the damage:
https://github.com/tgstation/tgstation/blob/e8476ae9336dcd7cccf65d89ecfaa86fa9330e4e/code/datums/mutations/touch.dm#L254-L255
It's _ALSO_ used as a mult on the healing cap itself, which _does_ affect the damage:
https://github.com/tgstation/tgstation/blob/e8476ae9336dcd7cccf65d89ecfaa86fa9330e4e/code/datums/mutations/touch.dm#L247-L250
https://github.com/tgstation/tgstation/blob/e8476ae9336dcd7cccf65d89ecfaa86fa9330e4e/code/datums/mutations/touch.dm#L256-L257

**Comment by 00-Steven** [comment]
The pacifist mult description makes sense under the assumption of the `heal_multiplier` and `pain_multiplier` comments:
https://github.com/tgstation/tgstation/blob/e8476ae9336dcd7cccf65d89ecfaa86fa9330e4e/code/datums/mutations/touch.dm#L155-L159
But doesn't knowing this essentially makes it deal _more_ damage than you heal.

# TL;DR
This is an issue of it multiplying the `heal_multiplier` and `pain_multiplier` vars on the `/datum/action/cooldown/spell/touch/lay_on_hands` action for each time it has been used, leading to increasing damage counts.

Additionally, contrary to the comments on `heal_multiplier` and `pain_multiplier`, `heal_multiplier` multiplies the healing AND damage (by multiplying both the base heal/damage AND the resulting heal). This makes the Pacifist bonus deal more than what it heals, contrary to what it sounds like it should do.

**Comment by carlarctg** [comment]
Second I saw this issue i added it as a fix on my pr but i forgor to link and desc it

**Comment by carlarctg** [comment]
Actually tyhat fix sucks adn this is a bit of a pain to handle
